pkg/clusteragent/clusterchecks/dispatcher_rebalance.go

```diff
@@ -182,7 +184,7 @@ func (d *dispatcher) rebalance() {
 			}
 
 			pickedNodeName := pickNode(diffMap, sourceNodeName)
-			if diffMap[pickedNodeName]+checkWeight < diffMap[sourceNodeName] {
+			if diffMap[pickedNodeName]+checkWeight < int(float64(diffMap[sourceNodeName])*tolerationMargin) {
```

Comment:
And another comment here to call out the usage of tolerationMargin to lean towards stability over perfectly optimal balance.